### PR TITLE
Networking: establish ALPN

### PR DESCRIPTION
**Pull request by kim**
While not terribly useful for a multiplexing protocol, we establish this
now in order to allow for future major protocol upgrades.

### CODE REVIEW

File: librad/src/net/quic.rs [hunk 1]
```diff
@@ -37,6 +37,12 @@ use crate::{
     peer::{self, PeerId},
 };
 
+/// The ALPN protocol(s) for the radicle-link protocol stack.
+///
+/// Not currently of significance, but established in order to allow future
+/// major protocol upgrades.
+const ALPN: &[&[u8]] = &[b"rad/1"];
```
Comment (xla) [inline librad/src/net/quic.rs:44]: Still not quite clear what the proper addressation of the protocol is. Is `link` just the name of the software artifact?

**Review by xla — APPROVED** [review]
🙃 🔊 😮 👜

**Review by FintanH — APPROVED** [review]
TIL

**Comment by kim** [comment]
> Still not quite clear what the proper addressation of the protocol is.
> Is `link` just the name of the software artifact?

`radicle-link` is the name of the protocol specification. That does not mean
that it has to be the URL scheme, URN namespace, or ALPN ID -- in all of those,
short identifiers are preferred, and the context is unambiguous.

**Comment by kim** [comment]
I considered `radl` for a moment...

**Comment by FintanH** [comment]
> I considered radl for a moment...

It's not too late :eyes:

**Comment by kim** [comment]
> It's not too late :eyes:

You may want to consult your Oktoberfest dictionary before +1-ing this :D

**Comment by xla** [comment]
> You may want to consult your Oktoberfest dictionary before +1-ing this :D

:beer: - If we work on `radl` does that make us `radler`? :drum:

**Pull request merged**